### ISSUE TITLE
多人通话回调问题

### ISSUE THREAD

**Issue by starise-wg** [ossrs/srs#2996]
**使用一对一通话或者多人通话时无法进行回调**

> 由于一对一或者多人通话时，因为网络突然中断或者相关不规范操作导致流异常退出，此时需要监控流是否存在，采用回调方式时通话流不会被识别为客户端，无法通过相关回调配置进行监控通话流，是否有别的方式实现通话流监控呢？
1. SRS版本(Version): `V4.0`

2. SRS的配置如下(Config):

```
vhost __defaultVhost__ {
 http_hooks {
        enabled         on;
        on_connect      https://5gvrtest.huanjianit.com/hjkj-video-center/srsCallback/onConnectCallBack;
        on_close        https://5gvrtest.huanjianit.com/hjkj-video-center/srsCallback/onCloseCallBack;
        on_publish      https://5gvrtest.huanjianit.com/hjkj-video-center/srsCallback/onPublicCallBack;
        on_unpublish    https://5gvrtest.huanjianit.com/hjkj-video-center/srsCallback/onUnpublicCallBack;
        on_play         https://5gvrtest.huanjianit.com/hjkj-video-center/srsCallback/onPlayCallBack;
        on_stop         https://5gvrtest.huanjianit.com/hjkj-video-center/srsCallback/onStopCallBack;
    }
    tcp_nodelay on;
    min_latency on;
    dvr {
        enabled on;
        dvr_path /data/[app]/[stream]/[2006]/[01]/[02]/[15]:[04]:[05].mp4;
        dvr_plan session;
        dvr_duration 10;
        dvr_wait_keyframe on;
        dvr_apply VideoManage/NK002 VideoManage/NK003 VideoManage/NK004 VideoManage/CSKS004 VideoManage/DSD002 VideoManage/DSD003 VideoManage/DSD004 VideoManage/CSKS001 VideoManage/CSKS002 VideoManage/CSKS003;
    }
```

**期望行为(Expect)**

> 监控相关通话流因为网络异常或者操作异常导致推出，然后调用踢流接口进行踢流操作

**Comment by ghost** [comment]
I am also looking for how to notify the WebRTC streaming client to refresh when the WebRTC streaming is stopped and restarted.

`TRANS_BY_GPT3`

**Comment by winlinvip** [comment]
This should not rely on a streaming media server, but should rely on its own business server, for example:

Each client establishes a connection with its own WebSocket server, so when it disconnects, your server is aware of it and can naturally notify other users.

In fact, IM (Instant Messaging) is a similar logic, and you can also choose an IM system to implement this functionality.

`TRANS_BY_GPT3`

**Comment by bbhxwl** [comment]
I deployed using Docker, set up an authentication callback function, but it seems to not be working even after restarting Docker. Why is that? Do I need to restart after modifying the configuration file?
![image](https://user-images.githubusercontent.com/26525599/175248060-fc28516a-4c32-488c-961e-5db05eb4f6ff.png)

`TRANS_BY_GPT3`

**Comment by starise-wg** [comment]
How do you say "look at the log"?



***@***.***
 
Sender: bbhxwl
Sent: 2022-06-23 16:02
Recipient: ossrs/srs
CC: wang.gang; Author
Subject: Re: [ossrs/srs] WebRTC: Handling and Callback Issues in Multi-Person Call Room (Issue #2996)
I deployed using Docker and set up an authentication callback function, but it seems to not be working even after restarting Docker. Why is that? Should I restart after modifying the configuration file, or...?
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

`TRANS_BY_GPT3`

**Comment by bbhxwl** [comment]
> How to view logs? ***@***.*** Sender: bbhxwl Sent: 2022-06-23 16:02 Recipient: ossrs/srs Cc: wang.gang; Author Subject: Re: [ossrs/srs] WebRTC: Handling and callback issues in multi-party call rooms (Issue #2996) I deployed using Docker and set up the authentication callback function, but it doesn't seem to work even after restarting Docker. Why is that? Do I need to restart after modifying the configuration file, or...? — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread. Message ID: ***@***.***>

Originally, the configuration file for Docker is not srs.conf. It is docker.conf.
Also, I would like to ask about the callback function. I am able to receive the request, but I cannot receive the parameters. Are the parameters sent via POST?

`TRANS_BY_GPT3`

**Comment by starise-wg** [comment]
***@***.***
 
Sender: bbhxwl
Sent: 2022-06-23 17:34
Recipient: ossrs/srs
CC: wang.gang; Author
Subject: Re: [ossrs/srs] WebRTC: Handling and Callback Issues in Multi-Person Call Rooms (Issue #2996)
How does the log say @.*** Sender: bbhxwl Sent: 2022-06-23 16:02 Recipient: ossrs/srs CC: wang.gang; Author Subject: Re: [ossrs/srs] WebRTC: Handling and Callback Issues in Multi-Person Call Rooms (Issue #2996) I deployed using Docker and set up the authentication callback function, but it doesn't seem to work. I also restarted Docker. Why is that? Do I need to restart after modifying the configuration file, or...? — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread. Message ID: @.***>
Originally, the Docker configuration file is not srs.conf. It is docker.conf.
Also, may I ask about the callback function? I can receive the request but not the parameters. Are the parameters sent via POST?
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

`TRANS_BY_GPT3`
